data_source.go

```diff
@@ -305,42 +342,74 @@ func (ds *pgDataSource) getPgCurrentWalLsn(role string) (string, error) {
 }
 
 func (ds *pgDataSource) getPgLastWalReplayLsn() (string, error) {
+	db, dbErr := ds.getDB()
+
+	if dbErr != nil {
+		return "", dbErr
+	}
+
 	pgLastWalLsn := null.String{}
-	err := ds.DB.Get(&pgLastWalLsn, "select pg_last_wal_replay_lsn()")
+	err := db.Get(&pgLastWalLsn, "select pg_last_wal_replay_lsn()")
 	if err != nil {
 		return "", err
 	}
 	return pgLastWalLsn.String, nil
 }
 
 func (ds *pgDataSource) getPgWalLsnDiff(currentLsn string, lastLsn string) (int64, error) {
+	db, dbErr := ds.getDB()
+
+	if dbErr != nil {
+		return 0, dbErr
+	}
+
 	var byteLag int64
 
 	query := fmt.Sprintf("select pg_wal_lsn_diff('%s', '%s')", currentLsn, lastLsn)
-	err := ds.DB.Get(&byteLag, query)
+
+	err := db.Get(&byteLag, query)
 	if err != nil {
 		return 0, err
 	}
 	return byteLag, nil
 }
 
 func (ds *pgDataSource) IsInRecovery() (bool, error) {
+	db, dbErr := ds.getDB()
+
```

Comment:
Nit-pick but for styling can we remove this extra line between `dbErr` and `dbErr != nil`. Few other instances in this file.